hnn_core/gui/gui.py

```diff
@@ -43,36 +45,51 @@
 }
 
 
-def _update_plot_window(simulation_data, _plot_out, plot_type,
-                        analysis_config):
-    """Refresh plots with data from simulation_data."""
+def _update_plot_window(simulation_data, analysis_config, plot_key):
+    """Refresh plots with data from simulation_data.
+
+    Parameters
+    ----------
+    simulation_data: dict
+        A dict of simulation data
+    analysis_config: dict
+        A dict of visualization configs
+    plot_key: str
+        A string used to identify the plot and data to update.
+    """
     # Make sure that visualization does not change the original data
-    dpls_copied = copy.deepcopy(simulation_data['dpls'])
-    net_copied = copy.deepcopy(simulation_data['net'])
+    _plot_out = analysis_config['plot_outputs'][plot_key]
```

Comment:
usually leading underscore in the variable name means that it is "private" or for internal use. Is that what you mean here? If not, maybe worth just removing the underscores.

Reply:
You're right I should remove the underscore there as it's safe to do so as it's already in a function. Will clean these sort of stuff later